src/qt/coincontroldialog.cpp

```diff
@@ -153,26 +165,194 @@ void CoinControlDialog::buttonBoxClicked(QAbstractButton* button)
+    {
+       ui->selectAllPushButton->setText("Select all");
+    }
+    else
+    {
+       ui->selectAllPushButton->setText("Select none");
+    }
+
+    CoinControlDialog::updateLabels(model, this);
+    showHideConsolidationReadyToSend();
+}
+
+void CoinControlDialog::buttonFilterModeClicked()
+{
+    if (m_FilterMode)
+    {
+        m_FilterMode = false;
+        ui->filterModePushButton->setText(">=");
+    }
+    else
     {
-        if (ui->treeWidget->topLevelItem(i)->checkState(COLUMN_CHECKBOX) != Qt::Unchecked)
+        m_FilterMode = true;
+        ui->filterModePushButton->setText("<=");
+    }
+}
+
+
+void CoinControlDialog::buttonFilterClicked()
+{
+    filterInputsByValue(m_FilterMode, ui->maxMinOutputValue->value(), 200);
+    showHideConsolidationReadyToSend();
+}
+
+bool CoinControlDialog::filterInputsByValue(const bool& less, const CAmount& inputFilterValue,
+                                            const unsigned int& inputCountLimit)
+{
+    QTreeWidgetItemIterator iter(ui->treeWidget);
+
+    // If less is true, then we are choosing the smallest inputs upward, and so the map comparator needs to be "less than".
+    // If less is false, then we are choosing the largest inputs downward, and so the map comparator needs to be "greater
+    // than".
+    auto comp = [less](CAmount a, CAmount b)
+    {
+        if (less)
         {
-            state = Qt::Unchecked;
-            break;
+            return (a < b);
         }
+        else
+        {
+            return (a > b);
+        }
+    };
+
+    std::multimap<CAmount, std::pair<QTreeWidgetItem*, COutPoint>, decltype(comp)> input_map(comp);
+
+    bool culled_inputs = false;
+
+    while (*iter)
+    {
+        CAmount input_value = (*iter)->text(COLUMN_AMOUNT_INT64).toLongLong();
+        COutPoint outpoint(uint256S((*iter)->text(COLUMN_TXHASH).toStdString()), (*iter)->text(COLUMN_VOUT_INDEX).toUInt());
+
+        if ((*iter)->text(COLUMN_TXHASH).length() == 64)
+        {
+            if ((*iter)->checkState(COLUMN_CHECKBOX) == Qt::Checked
+                    && ((less && input_value <= inputFilterValue) || (!less && input_value >= inputFilterValue)))
+            {
+                input_map.insert(std::make_pair(input_value, std::make_pair(*iter, outpoint)));
+            }
+            else
+            {
+                (*iter)->setCheckState(COLUMN_CHECKBOX, Qt::Unchecked);
+                coinControl->UnSelect(outpoint);
+            }
+        }
+
+        ++iter;
     }
-    ui->treeWidget->setEnabled(false);
-    for (int i = 0; i < ui->treeWidget->topLevelItemCount(); i++)
-            if (ui->treeWidget->topLevelItem(i)->checkState(COLUMN_CHECKBOX) != state)
-                ui->treeWidget->topLevelItem(i)->setCheckState(COLUMN_CHECKBOX, state);
-    ui->treeWidget->setEnabled(true);
-    CoinControlDialog::updateLabels(model, this);
+
+    // The second loop is to limit the number of selected outputs to the inputCountLimit.
```

Comment:
The original tests showed that consolidate transactions were failing above 200. It was also taking an extraordinary amount of time locking up the GUI because of the wallet core to GUI update triggers. The performance issue no longer exists because of the optimizaitons I did on the GUI update earlier, and this is originating in the GUI anyway. We could probably safely raise the output limits for both the rpc and here to 600.

Reply:
We could also do it based on size, although that is an estimate and is not completely reliable.

Reply:
I would rather not allow the filter to permit more than the number of outputs that could be successfully sent. I think the right answer is to raise the output limit fixed value.

Reply:
Ugh. After looking at this, I agree with you. I have removed the output count limitation from the filter button.